realm/realm-library/src/androidTest/kotlin/io/realm/UUIDTests.kt

```diff
@@ -0,0 +1,590 @@
+    fun queriesLessThan() {
+        val uuid1 = UUID.fromString("017ba5ca-aa12-4afa-9219-e20cc3018599")
+        val uuid2 = UUID.fromString("027ba5ca-aa12-4afa-9219-e20cc3018599")
+        val uuid3 = UUID.fromString("037ba5ca-aa12-4afa-9219-e20cc3018599")
+
+        realm.executeTransaction { realm ->
+            realm.createObject<UUIDAndString>().id = uuid2
+            realm.createObject<UUIDAndString>().id = null
+            realm.createObject<UUIDAndString>().id = uuid3
+            realm.createObject<UUIDAndString>().id = uuid1
+        }
+
+        val all = realm.where<UUIDAndString>()
+                .lessThan("id", uuid2)
+                .sort("id", Sort.ASCENDING)
+                .findAll()
+
+        assertEquals(1, all.size)
+        assertEquals(uuid1, all[0]!!.id)
+    }
+
+    @Test
+    fun queriesIsNull() {
+        val uuid1 = UUID.randomUUID()
+        val uuid2 = UUID.randomUUID()
+        val uuid3 = UUID.randomUUID()
+
+        realm.executeTransaction { realm ->
+            realm.createObject<UUIDAndString>().id = uuid2
+            realm.createObject<UUIDAndString>().id = null
+            realm.createObject<UUIDAndString>().id = uuid3
+            realm.createObject<UUIDAndString>().id = uuid1
+        }
+
+        val all = realm.where<UUIDAndString>()
+                .isNull("id")
+                .findAll()
+
+        assertEquals(1, all.size)
+        assertNull(all[0]!!.id)
+    }
+
+    @Test
+    fun queriesIsNotNull() {
+        val uuid1 = UUID.fromString("017ba5ca-aa12-4afa-9219-e20cc3018599")
+        val uuid2 = UUID.fromString("027ba5ca-aa12-4afa-9219-e20cc3018599")
+        val uuid3 = UUID.fromString("037ba5ca-aa12-4afa-9219-e20cc3018599")
+
+        realm.executeTransaction { realm ->
+            realm.createObject<UUIDAndString>().id = uuid2
+            realm.createObject<UUIDAndString>().id = null
+            realm.createObject<UUIDAndString>().id = uuid3
+            realm.createObject<UUIDAndString>().id = uuid1
+        }
+
+        val all = realm.where<UUIDAndString>()
+                .isNotNull("id")
+                .sort("id", Sort.ASCENDING)
+                .findAll()
+
+        assertEquals(3, all.size)
+        assertEquals(uuid1, all[0]!!.id)
+        assertEquals(uuid2, all[1]!!.id)
+        assertEquals(uuid3, all[2]!!.id)
+    }
+
+    @Test
+    fun queriesAverage() {
+        val uuid1 = UUID.randomUUID()
+        val uuid2 = UUID.randomUUID()
+        val uuid3 = UUID.randomUUID()
+
+        realm.executeTransaction { realm ->
+            realm.createObject<UUIDAndString>().id = uuid2
+            realm.createObject<UUIDAndString>().id = null
+            realm.createObject<UUIDAndString>().id = uuid3
+            realm.createObject<UUIDAndString>().id = uuid1
+        }
+
+        assertFailsWith<IllegalArgumentException>("Average is not supported for UUID") {
+            realm.where<UUIDAndString>().average("id") // FIXME should we support average queries in Core?
```

Comment:
I think we can remove this FIXME.  Aggregate methods like `average` doesn't really make sense for UUID's.